utils/ensureTrailingSlash.ts

```diff
@@ -0,0 +1,3 @@
+export function ensureTrailingSlash(str: string): string {
+    return str.replace(/\/+$/, '') + '/';
```

Comment:
```suggestion
    return str.endsWith('/') ? str : str + '/';
```

A URL such as `https://flagsmith.example.com////` is technically distinct from `https://flagsmith.example.com/`, and this regex would remove all trailing slashes except one. I know this sounds like extreme bikeshedding but it's closer in line to what we do in other SDKs, e.g. [Go](https://github.com/Flagsmith/flagsmith-go-client/blob/3825ef5f28749320e36174dfcaa70eb30e2ab3f5/options.go#L143), [.NET](https://github.com/Flagsmith/flagsmith-dotnet-client/blob/4f760b982ded4559a768b8933a4388b9e5690341/Flagsmith.FlagsmithClient/UrlExtensions.cs#L12), [Node.js](https://github.com/Flagsmith/flagsmith-nodejs-client/blob/2d26062619ec6f2855211551c734bf09cf9897b8/sdk/index.ts#L141), [Ruby](https://github.com/Flagsmith/flagsmith-ruby-client/blob/76bc72289ffd211efb699f79713659fe1b38cb59/lib/flagsmith/sdk/config.rb#L106)